CONTRIBUTING.md

```diff
@@ -0,0 +1,370 @@
+create a feature branch for making your changes. For example:
+
+    git branch shiny-new-feature
+    git checkout shiny-new-feature
+
+The above can be simplified to:
+
+    git checkout -b shiny-new-feature
+
+This changes your working directory to the shiny-new-feature branch.
+Keep any changes in this branch specific to one bug or feature so it is
+clear what the branch brings to *UV-CDAT*. You can have many
+shiny-new-features and switch in between them using the git checkout
+command.
+
+### Making changes
+Before making your code changes, it is often necessary to build the code
+that was just checked out. The best way to develop *UV-CDAT* is to build
+using default settings:
+
+        mkdir uvcdat-build
+        cmake uvcdat-path-to-source
+        make -jN
+
+Contributing to the documentation
+---------------------------------
+
+If you're not the developer type, contributing to the documentation is
+still of huge value. You don't even have to be an expert on *UV-CDAT* to
+do so! Something as simple as pointing missing information or broken links
+will be of great value.
+
+Contributing to the code base
+-----------------------------
+### Code Standards
+
+*UV-CDAT* uses the [PEP8](http://www.python.org/dev/peps/pep-0008/)
+standard. There are several tools to ensure you abide by this standard.
+
+Alternatively, use [flake8](http://pypi.python.org/pypi/flake8) tool for
+checking the style of your code. Additional standards are outlined on
+the [code style wiki
+page](LINK HERE).
+
+Please try to maintain backward-compatibility. *UV-CDAT* has lots of
+users with lots of existing code, so don't break it if at all possible.
+If you think breakage is required clearly state why as part of the Pull
+Request. Also, be careful when changing method signatures and add
+deprecation warnings where needed.
+
+### Test-driven Development/Writing Code
+*UV-CDAT* is serious about [Test-driven Development
+(TDD)](http://en.wikipedia.org/wiki/Test-driven_development). This
+development process "relies on the repetition of a very short
+development cycle: first the developer writes an (initially failing)
+automated test case that defines a desired improvement or new function,
+then produces the minimum amount of code to pass that test." So, before
+actually writing any code, you should write your tests. Often the test
+can be taken from the original GitHub issue. However, it is always worth
+considering additional use cases and writing corresponding tests.
+
+Adding tests is one of the most common requests after code is pushed to
+*UV-CDAT*. It is worth getting in the habit of writing tests ahead of
+time so this is never an issue.
+
+#### Writing tests
+
+All tests should go into the *tests* subdirectory of the specific
+package. There are probably many examples already there and looking to
+these for inspiration is suggested.
+
+The `testing.checkimage.py` module has special `check_result_image` function
```

Comment:
Sure, I was thinking for later. Currently most of the tests are regression testing so no need for separate section. But once we have more standard test mechanism, we can move them into sub-sections.